### PR TITLE
Screen size message

### PR DESCRIPTION
**Pull request by odeimaiz**
## What do these changes do?

Pops up an alert message when browser's resolution goes below 1240x700

SmallScreen

### CODE REVIEW

**Review by mguidon — COMMENTED** [review]
What about a shut up and dont show me again?

**Comment by pcrespov** [comment]
Is there a specific reason why you are adding this?

IMO this should be added as a "TIP" message  and the UX should make you feel it that way, e.g.
- I would add "TIP" in the header message. 
- instead of a modal window, it should be a message that appears (e.g. from top) and dissapears after a time or can be closed. 
- It also appears only once
- It has a check-box that says "Dont' show me again"
- Settings include a section to reset these
- The message should not be that direct as: "Get a bigger device" but rather "For optimal UX, we highly recommend ...."

**Review by pcrespov — COMMENTED** [review]
Is there a specific reason why you are adding this?

IMO this should be added as a "TIP" message  and the UX should make you feel it that way, e.g.
- I would add "TIP" in the header message. 
- instead of a modal window, it should be a message that appears (e.g. from top) and dissapears after a time or can be closed. 
- It also appears only once
- It has a check-box that says "Dont' show me again"
- Settings include a section to reset these
- The message should not be that direct as: "Get a bigger device" but rather "For optimal UX, we highly recommend ...."

**Comment by odeimaiz** [comment]
> Is there a specific reason why you are adding this?
> 
> IMO this should be added as a "TIP" message and the UX should make you feel it that way, e.g.
> 
> * I would add "TIP" in the header message.
> * instead of a modal window, it should be a message that appears (e.g. from top) and dissapears after a time or can be closed.
> * It also appears only once
> * It has a check-box that says "Dont' show me again"
> * Settings include a section to reset these options
> * The message should not be that direct as: "Get a bigger device" but rather "For optimal UX, we highly recommend ...."

We got some requests from users with small screens saying that the app doesn't look very good or some buttons are out of view. We thought that showing a message when this happens would be useful.
It can, of course, be closed by pressing the only available button that says "Got it".
I don't agree with the "Don't show me again" approach.
The title was updated to "Oops, the window is a bit too small!"

File: services/web/client/source/class/osparc/Application.js [hunk 1]
```diff
@@ -88,15 +87,62 @@ qx.Class.define("osparc.Application", {
...
       this.__checkCookiesAccepted();
+
+      // onload, load, DOMContentLoaded, appear... didn't work
+      // bit of a hack
+      setTimeout(() => this.__checkScreenSize(), 100);
+      window.addEventListener("resize", () => this.__checkScreenSize());
+    },
+
+    __checkScreenSize: function() {
+      const title = this.tr("Oops, the window is a bit too small!");
```
Comment (GitHK) [inline services/web/client/source/class/osparc/Application.js:105]:
```suggestion
      const title = this.tr("Screen resolution or window size is bit too small!");
```

File: services/web/client/source/class/osparc/Application.js [hunk 2]
```diff
@@ -88,15 +87,62 @@ qx.Class.define("osparc.Application", {
...
+        modal: true,
+        contentPadding: 10
+      });
+      const w = document.documentElement.clientWidth;
+      const h = document.documentElement.clientHeight;
+      if (this.self().MIN_WIDTH > w || this.self().MIN_HEIGHT > h) {
+        const msg = this.tr(`
+          oSPARC is designed for slightly bigger screens.<br>\
+          A mininum resolution of ${this.self().MIN_WIDTH}x${this.self().MIN_HEIGHT} is recommended<br>\
+          Touch devices are not fully supported.
```
Comment (GitHK) [inline services/web/client/source/class/osparc/Application.js:124]:
```suggestion
          oSPARC is designed for slightly bigger screens (window size).<br>\
          A mininum resolution (window size) of ${this.self().MIN_WIDTH}x${this.self().MIN_HEIGHT} is recommended<br>\
          Touch devices are not fully supported.
```

**Review by GitHK — APPROVED** [review]
I would improve the message to be more user friendly.

**Review by KZzizzle — APPROVED** [review]

**Pull request merged**